benchmark-framework/src/main/java/io/openmessaging/benchmark/worker/DistributedWorkersEnsemble.java

```diff
@@ -245,54 +235,31 @@ public PeriodStats getPeriodStats() {
-        individualStats.forEach(is -> {
-            try {
-                stats.publishLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                        ByteBuffer.wrap(is.publishLatencyBytes), TimeUnit.SECONDS.toMicros(30)));
-            } catch (Exception e) {
-                log.error("Failed to decode publish latency");
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.scheduleLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                    ByteBuffer.wrap(is.scheduleLatencyBytes), TimeUnit.SECONDS.toMicros(30)));
-            } catch (Exception e) {
-                log.error("Failed to decode schedule latency");
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.publishDelayLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                        ByteBuffer.wrap(is.publishDelayLatencyBytes), TimeUnit.SECONDS.toMicros(30)));
-            } catch (Exception e) {
-                log.error("Failed to decode publish delay latency: {}",
-                          ByteBufUtil.prettyHexDump(Unpooled.wrappedBuffer(is.publishDelayLatencyBytes)));
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.publishDelayLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                        ByteBuffer.wrap(is.publishDelayLatencyBytes), TimeUnit.SECONDS.toMicros(30)));
-            } catch (Exception e) {
-                log.error("Failed to decode publish delay latency: {}",
-                          ByteBufUtil.prettyHexDump(Unpooled.wrappedBuffer(is.publishDelayLatencyBytes)));
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.endToEndLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                        ByteBuffer.wrap(is.endToEndLatencyBytes), TimeUnit.HOURS.toMicros(12)));
-            } catch (Exception e) {
-                log.error("Failed to decode end-to-end latency: {}",
-                        ByteBufUtil.prettyHexDump(Unpooled.wrappedBuffer(is.endToEndLatencyBytes)));
-                throw new RuntimeException(e);
-            }
-        });
+        final CumulativeLatencies stats = new CumulativeLatencies();
+        individualStats.forEach(is -> Map.of(
+                        stats.publishLatency, is.publishLatencyBytes,
+                        stats.scheduleLatency, is.scheduleLatencyBytes,
+                        stats.publishDelayLatency, is.publishDelayLatencyBytes,
+                        stats.endToEndLatency, is.endToEndLatencyBytes)
+                .forEach((histogram, bytes) -> {
+                    final ByteBuffer buffer = ByteBuffer.wrap(bytes);
+                    try {
+                        histogram.add(Histogram.decodeFromCompressedByteBuffer(buffer,
+                                TimeUnit.SECONDS.toMicros(30)));
```

Comment:
at least the last one uses 12 hours not 30 seconds in the list above, so this maybe needs to be variable too?